src/backend/distributed/executor/multi_executor.c

```diff
@@ -66,7 +66,8 @@ static CustomExecMethods RouterSingleModifyCustomExecMethods = {
 	.ExplainCustomScan = CitusExplainScan
 };
 
-static CustomExecMethods RouterMultiModifyCustomExecMethods = {
+/* not static to enable swapping in multi-modify logic during router execution */
+CustomExecMethods RouterMultiModifyCustomExecMethods = {
```

Comment:
It seems the only distinction between `RouterSingleModifyCustomExecMethods` and `RouterMultiModifyCustomExecMethods` is in the `ExecCustomScan` and the only metric used to choose between them is the number of tasks. What if we collapse them into one `ExecMethods` with a `RouterModifyExecScan` that calls the right function based on the task count?